toolkit/tools/pkg/imagecustomizerlib/customizeuki.go

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
+)
+
+func prepareUki(uki *imagecustomizerapi.Uki, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	if uki == nil {
+		return nil
+	}
+
+	logger.Log.Infof("Enable uki")
+
+	// Check UKI dependency packages.
+	err = validateUkiDependencies(imageChroot)
+	if err != nil {
+		return fmt.Errorf("failed to validate package dependencies for uki:\n%w", err)
+	}
+
+	// Create necessary directories for UKI.
+	err = createUkiDirectories(imageChroot)
+	if err != nil {
+		return fmt.Errorf("failed to create UKI directories:\n%w", err)
+	}
+
+	// Install systemd-boot.
+	err = imageChroot.UnsafeRun(func() error {
+		return shell.ExecuteLiveWithErr(1, "sudo", "bootctl", "install", "--no-variables")
+	})
+	if err != nil {
+		return fmt.Errorf("failed to install systemd-boot:\n%w", err)
+	}
+
+	// Copy UKI specific files.
+	err = copyUkiFiles(imageChroot)
+	if err != nil {
+		return fmt.Errorf("failed to copy UKI files:\n%w", err)
+	}
+
+	// Prepare ukify config files.
+	bootDir := filepath.Join(imageChroot.RootDir(), "/boot")
+
+	kernels, err := findKernelImages(bootDir)
+	if err != nil {
+		return err
+	}
+
+	err = ensureInitramfsForKernels(bootDir, kernels)
+	if err != nil {
+		return err
+	}
+
+	for _, kernel := range kernels {
+		err = createUkifyConfig(bootDir, kernel)
+		if err != nil {
+			return fmt.Errorf("failed to create ukify config: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func validateUkiDependencies(imageChroot *safechroot.Chroot) error {
+	requiredRpms := []string{"systemd-ukify", "systemd-boot", "efibootmgr"}
+
+	// Iterate over each required package and check if it's installed.
+	for _, pkg := range requiredRpms {
+		logger.Log.Debugf("Checking if package (%s) is installed", pkg)
+		if !isPackageInstalled(imageChroot, pkg) {
+			return fmt.Errorf("package (%s) is not installed:\nthe following packages must be installed to use Uki: %v", pkg, requiredRpms)
```

Comment:
Why is systemd-boot not treated the same way as these other packages (not saying it should be, but would be good to document why).